core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -558,10 +558,10 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
    * @throws IllegalStateException if a partition is not in `partitionsBeingReassigned`
    */
   private def maybeTriggerPartitionReassignment(topicPartitions: Set[TopicPartition]) {
-    topicPartitions.foreach { tp =>
+    val partitionsToBeRemovedFromReassignment = topicPartitions.filter { tp =>
```

Comment:
@ijuma Ah I just realized that I missed your comment.

Thanks much for the comment. This certainly makes sense. I have updated the patch using the first approach as suggested. The second approach will be a bit more complex because a partition needs to be removed from assignment if onPartitionReassignment(..) throws exception.